### PR TITLE
Add support of dataset embedding cache for Animatediff training

### PR DESCRIPTION
**Pull request by Gaohan123**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

**Review by wtomin — APPROVED** [review]
LGTM.

File: examples/animatediff/args_train.py [hunk 1]
```diff
@@ -170,12 +170,22 @@ def parse_args():
     parser.add_argument(
         "--motion_module_path", default="", type=str, help="path to pretrained motion mdule. Load it if not empty"
     )
+    parser.add_argument("--train_data_type", default="video_file", type=str, help="type of data for training")
```
Comment (HaoyangLee) [inline examples/animatediff/args_train.py:173]: add `choices=[]` to clarify all the possible train_data_type

Comment (Gaohan123) [reply examples/animatediff/args_train.py]: fixed

**Review by HaoyangLee — APPROVED** [review]

**Review by hadipash — APPROVED** [review]

File: examples/animatediff/ad/models/diffusion/ddpm.py
Comment (SamitHuang) [inline examples/animatediff/ad/models/diffusion/ddpm.py:1]: Decouple the embedding cache diffusion and normal diffusion for clarity. e.g. 
```
class LatentDiffusionWithEmbedding(LatentDiffusion):
   ...
```
and overwrite the `get_condition_embeddings` and `get_latents` methods.

And invoke this Diffusion class in an independent config yaml file

File: examples/animatediff/ad/data/dataset.py
Comment (SamitHuang) [inline examples/animatediff/ad/data/dataset.py:1]: Similarly, it's better to decouple the dataset class for embedding data from the original one.

File: examples/animatediff/ad/models/diffusion/ddpm.py
Comment (Gaohan123) [reply examples/animatediff/ad/models/diffusion/ddpm.py:1]: fixed

File: examples/animatediff/ad/data/dataset.py
Comment (Gaohan123) [reply examples/animatediff/ad/data/dataset.py:1]: fixed

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**